### PR TITLE
Fix zombie process problem

### PR DESCRIPTION
**Pull request by CcydtN**
Describe your PR, what does it fix/add?
_Fixing the problem about zombie process_

Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
_This is the same method used by sway. Solving the problem by forking a child and grandchild. Then, force the grandchild become orphaned, so it will be handled by the init process._

Is it ready for merging, or does it need work?
_It should be ready for merging_

### CODE REVIEW

**Comment by vaxerski** [comment]
let's just say i got no clue why the fuck that's needed but iirc sway did it similiarly.

**Comment by vaxerski** [comment]
if you could just fix the styling it would be cool, stuff like

```
if (cond) {

}
```

regular indentation,

```
int a, b;
```

etc etc

File: src/config/ConfigManager.cpp [hunk 1]
```diff
@@ -169,12 +169,48 @@ void CConfigManager::handleRawExec(const std::string& command, const std::string
         toExec = std::string("WAYLAND_DISPLAY=") + std::string(g_pCompositor->m_szWLDisplaySocket) + " " + toExec;
 
     Debug::log(LOG, "Config executing %s", toExec.c_str());
-
-    if (fork() == 0) {
-        execl("/bin/sh", "/bin/sh", "-c", toExec.c_str(), nullptr);
-
+    
+    int socket[2];
+	if (pipe(socket) != 0) {
```
Comment (vaxerski) [inline src/config/ConfigManager.cpp:174]: indent

File: src/config/ConfigManager.cpp [hunk 2]
```diff
@@ -169,12 +169,48 @@ void CConfigManager::handleRawExec(const std::string& command, const std::string
...
+		Debug::log(LOG, "Unable to create pipe for fork");
+	}
+
+    pid_t child ,grandchild;
```
Comment (vaxerski) [inline src/config/ConfigManager.cpp:178]: style

File: src/config/ConfigManager.cpp [hunk 3]
```diff
@@ -169,12 +169,48 @@ void CConfigManager::handleRawExec(const std::string& command, const std::string
...
+    child = fork();
+    if (child<0){
```
Comment (vaxerski) [inline src/config/ConfigManager.cpp:180]: spacing

File: src/config/ConfigManager.cpp [hunk 4]
```diff
@@ -169,12 +169,48 @@ void CConfigManager::handleRawExec(const std::string& command, const std::string
...
+        if (grandchild==0){
+            // run in grandchild
+            close(socket[0]);
+            close(socket[1]);
+            execl("/bin/sh", "/bin/sh", "-c", args.c_str(), nullptr);
+            // exit grandchild
+            _exit(0);
+        }
+        close(socket[0]);
+        (void) write(socket[1], &grandchild, sizeof(grandchild));
```
Comment (vaxerski) [inline src/config/ConfigManager.cpp:198]: (void) redundant

**Review by vaxerski — COMMENTED** [review]
not all instances highlighted, but examples of poor styling.

**Comment by vaxerski** [comment]
ok you formatted too much, you raped the format outside that one func too. Please do not autoformat the entire file.

Only your func should've had its format fixed.

**Comment by CcydtN** [comment]
Ops, sorry for that.
I redo the format fix and everything should be fine now.
Please check if there any other problem.

**Comment by vaxerski** [comment]
@fufexan can you verify plox xoxo

**Comment by fufexan** [comment]
Tbh idk what to verify, I don't even see the zombie processes in `btm`. Will check with `htop`.

**Comment by fufexan** [comment]
@CcydtN well it seems like it's not completely fixed.
image

**Comment by CcydtN** [comment]
> @CcydtN well it seems like it's not completely fixed.
> image
> 

This pic didnt show any zombie process, the process in lime are threads.

You can click on S column or choose F6 (SortBy) and then select STATE. And click "capital i" to sort it in descending order.

**Comment by CcydtN** [comment]
> if it spawns, it works

@vaxerski Do you mean "NOT spawns"?
Preventing the zombie process is only one purpose of this pull request, and I think this line make @fufexan confused.

**Comment by vaxerski** [comment]
we clarified that on the discord server dw

**Comment by vaxerski** [comment]
so, @fufexan is everything alr? no zombies?

**Comment by fufexan** [comment]
Will have to check again. One sec.

**Comment by vaxerski** [comment]
can you do what @CcydtN said to hide threads?

**Comment by CcydtN** [comment]
@fufexan
"F6" is for sorting.
This option on the "F2" menu should hide the thread.
image

**Comment by fufexan** [comment]
Then no, there are no other processes for Hyprland.

**Comment by vaxerski** [comment]
ok

**Pull request merged**